sklearn/cluster/tests/test_optics.py

```diff
@@ -220,7 +220,7 @@ def test_nowarn_if_metric_bool_data_bool():
     # https://github.com/scikit-learn/scikit-learn/issues/18996
 
     pairwise_metric = 'rogerstanimoto'
-    X = np.random.randint(2, size=(5, 2), dtype=np.bool)
+    X = np.random.randint(2, size=(5, 2), dtype=np.bool_)
```

Comment:
I would prefer:

```suggestion
    X = np.random.randint(2, size=(5, 2), dtype=bool)
```

As suggested in the warning. Indeed, is the approach that we had followed in similar PR: #18927, #17687, #17707.